File: VetPlatform.UI/src/app/models/bookings.response.model.ts
import { ResponseModel } from './response.model';

export interface BookingsResponseModel extends ResponseModel {
    payload: Booking[];
    hasMore: boolean;
}

export interface Booking {
    id:string,
    schedule: Date,
    scheduleTime: string,
    status: string,
    customer: Customer
}

export interface Customer {
    id: string,
    name:string
}

File: VetPlatform/Models/BookingsResponseModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace VetPlatform.Api.Models
{
    
    public class BookingsResponseModel
    {
        public bool Success { get; set; }
        public bool HasMore { get; set; }
        public List<BookingResponseModel> Payload { get; set; }
    }


    public class BookingResponseModel
    {
        public Guid Id { get; set; }
        public CustomerResponseModel Customer { get; set; }
        public DateTime Schedule { get; set; }
        public string ScheduleTime => Schedule.ToString("HH:mm");

        public string Status { get; set; }
    }

    public class CustomerResponseModel
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
    }
}
